nodes.go

```diff
@@ -620,10 +621,9 @@ func (s *NodesService) putOrOverwrite(in io.Reader, httpVerb, url, name, metadat
 	contentType := writer.FormDataContentType()
 	contentLength := int64(-1)
 
-	buf := make([]byte, 1)
-	n, err := in.Read(buf)
-	isZeroLength := n == 0 && err == io.EOF
-	in = io.MultiReader(bytes.NewReader(buf), in)
+	bufIn := bufio.NewReader(in)
```

Comment:
It indeed does copy, but only if the read requests are smaller than the buffer:
https://golang.org/src/bufio/bufio.go#L182

At your choosing, I can either make this a bufio with buffer size 1 byte or use the previous (but fixed) approach.